CareKitUI/CareKitUI/iOS/Task/SliderTaskView.swift

```diff
@@ -0,0 +1,243 @@
+    private let isSliderViewPadded: Bool
+    private let header: Header
+    private let sliderView: SliderView
+    private let instructions: Text?
+    
+    public var body: some View {
+        CardView {
+            VStack(alignment: .leading, spacing: style.dimension.directionalInsets1.top) {
+                VStack { header }
+                    .if(isCardEnabled && isHeaderPadded) { $0.padding([.horizontal, .top]) }
+                
+                instructions?
+                    .font(.subheadline)
+                    .fontWeight(.medium)
+                    .lineLimit(nil)
+                    .if(isCardEnabled) { $0.padding([.horizontal]) }
+                
+                VStack { sliderView }
+                    .if(isCardEnabled && isHeaderPadded) { $0.padding([.horizontal, .bottom]) }
+            }
+        }
+    }
+    
+    // MARK: - Init
+    
+    private init(isHeaderPadded: Bool, isSliderViewPadded: Bool,
+                 instructions: Text?, @ViewBuilder header: () -> Header, @ViewBuilder sliderView: () -> SliderView) {
+        self.isHeaderPadded = isHeaderPadded
+        self.isSliderViewPadded = isSliderViewPadded
+        self.instructions = instructions
+        self.header = header()
+        self.sliderView = sliderView()
+    }
+    
+    /// Create an instance.
+    /// - Parameter instructions: Instructions text to display under the header.
+    /// - Parameter header: Header to inject at the top of the card. Specified content will be stacked vertically.
+    /// - Parameter sliderView: View to inject under the header. Specified content will be stacked vertically.
+    public init(instructions: Text? = nil, @ViewBuilder header: () -> Header, @ViewBuilder sliderView: () -> SliderView) {
+        self.init(isHeaderPadded: false, isSliderViewPadded: false, instructions: instructions, header: header, sliderView: sliderView)
+    }
+}
+
+public extension SliderTaskView where Header == _SliderTaskViewHeader {
+    
+    /// Create an instance.
+    /// - Parameter title: Title text to display in the header.
+    /// - Parameter detail: Detail text to display in the header.
+    /// - Parameter instructions: Instructions text to display under the header.
+    /// - Parameter sliderView: View to inject under the header. Specified content will be stacked vertically.
+    init(title: Text, detail: Text? = nil, instructions: Text? = nil, @ViewBuilder sliderView: () -> SliderView) {
+        self.init(isHeaderPadded: true, isSliderViewPadded: false, instructions: instructions, header: {
+            _SliderTaskViewHeader(title: title, detail: detail)
+        }, sliderView: sliderView)
+    }
+}
+
+public extension SliderTaskView where SliderView == _SliderTaskViewSliderView {
+    
+    /// Create an instance.
+    /// - Parameter instructions: Instructions text to display under the header.
+    /// - Parameter isComplete; True if the button under the slider is in the completed.
```

Comment:
What effect does this parameter have on the view? This view allows you to log multiple outcome values for a single task, so does it matter if the task is completed, given you can always log additional outcomes?

Reply:
Originally, I did not have logging in mind as its primary function. I removed `isComplete` from all of the Slider task files.